lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
```

Comment:
could probably simplify to just state that dotdirs (but not dotfiles) in the root are magically excluded even if they're not chefignored.